ec/ecresource/deploymentresource/deployment/v2/deployment_update_payload.go

```diff
@@ -60,9 +60,11 @@ func (plan DeploymentTF) getBaseUpdatePayloads(ctx context.Context, client *api.
 		Kibana:             template.DeploymentTemplate.Resources.Kibana,
 	}
 
-	// If the deployment template has changed then we should use the template migration API
-	// to build the base update payloads
-	if newDtId != prevDtId && prevDtId != "" {
+	// If the deployment template has changed or MigrateToLatestHardware is true, we should use the template migration
+	// API to build the base update payloads
+	migrateToLatest := plan.MigrateToLatestHardware.ValueBool() || (newDtId != prevDtId && prevDtId != "")
```

Comment:
Does it make sense to check that migration is available (as the plan modifier does) in this new case (when `MigrateToLatestHardware` is set) before calling migration API?

Reply:
Also, I think we can do some optimisations for this new case (only when DT isn't modified) - `Update` always follows `Read` so we can store `migrateUpdateRequest` in `ReadRequest.Private` in `Read` and then get it from `UpdateRequest.Private` in `Update` using `ec/ecresource/deploymentresource/private_state.go`. However, feel free to merge without implementing this now.

Reply:
Thanks for the suggestions 🙏 

> Does it make sense to check that migration is available

Done in https://github.com/elastic/terraform-provider-ec/pull/755/commits/fa8a4499d789a7752a83623d64fffcf0b87671b3.

> I think we can do some optimisations for this new case

Done in https://github.com/elastic/terraform-provider-ec/pull/755/commits/a23377791b701b46b33ccbce8ebdefaae2073b15.